activerecord/lib/active_record/associations/has_one_association.rb

```diff
@@ -59,7 +59,7 @@ def delete(method = options[:dependent])
             target.delete
           when :destroy
             target.destroyed_by_association = reflection
-            target.destroy
+            throw(:abort) if target.destroy == false
```

Comment:
I think this would be safer without the `==` - just do `throw(:abort) unless target.destroy`

Reply:
`throw(:abort) unless target.destroy` doesn't work. I'm still investigating why. Perhaps something wrong with destroy method?

Reply:
The only reason that may not work is if you want to differentiate between `false` and `nil` in the return of `destroy`.

Reply:
in `activerecord/lib/activerecord/callbacks.rb` in the destroy method, there's a rescue for `RecordNotDestroyed` that returns false (not nil). The method might return nil for other reasons.